semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -43,64 +49,107 @@ module H = Parse_mini_rule
 (* Helpers *)
 (*****************************************************************************)
 
-(* less: could use a hash to accelerate things *)
-let rec find_fields flds xs =
-  match flds with
-  | [] -> ([], xs)
-  | fld :: flds ->
-      let fld_match = List.assoc_opt fld xs in
-      let xs = List.remove_assoc fld xs in
-      let matches, rest = find_fields flds xs in
-      ((fld, fld_match) :: matches, rest)
+type env = { id : string; languages : R.xlang; path : string list }
 
 let error s = raise (E.InvalidYamlException s)
 
-let rec yaml_to_json = function
-  | `Null -> J.Null
-  | `Bool b -> J.Bool b
-  | `Float f ->
-      (* or use J.Int? *)
-      J.Float f
-  | `String s -> J.String s
-  | `A xs -> J.Array (xs |> List.map yaml_to_json)
-  | `O xs -> J.Object (xs |> List.map (fun (k, v) -> (k, yaml_to_json v)))
+let generic_to_json key ast =
+  let rec generic_to_json_help = function
```

Comment:
It's quite common to name this kind of auxiliary functions as `aux`, `loop`, `go`, or something like that.

Reply:
done